### DISCUSSION TITLE
q-select with deep nested objects

### DISCUSSION THREAD

**Discussion by jaymc-arg** [quasarframework/quasar#14229] [General - Components / Directives / etc] [answered]
Hi,
I'm trying to acces a deep nested object with `:option-label` but i can not get it working.
This is my object:
```js
[
  {
     prop1: "asd,
     prop2: [
         {
            prop3: "ASD"
         }
    ]
  }
]
```
I want to render the `another_prop`. I'm trying `:option-label="(item) => item.prop2[0].prop3"` but it tells me it "can not read '0' of Undefined".
The thing is that if i try to render `:option-label="(item) => item.prop2` it shows the options for `[Object][Object]` so `prop2` is not `Undefined`

Can someone help me here?

Thanks in advance.

**Comment by MilosPaunovic** [comment]
It tells you that it `can not read '0' of undefined` cause the `prop1` is not an array.
What you could do is create a computed property which will return uniformed structure to `QSelect`.

**Reply by jaymc-arg** [reply]
I guess for your answer there is no way to do with a q-select option, but just to clarify i want to access `prop3` inside `prop2` which is actually an array.

**Comment by pdanpdan** [comment]
It should be no problem. Make your function check if prop2 is array

**Reply by jaymc-arg** [reply]
If I try to `console.log` the object I can access prop3 like this `refObj.value.array[0].prop2[0].prop3` so it's an array.
Also 
```js
refObj.value.array.forEach((element) => {
        console.log(element.prop2[0].prop3);
      })
```
works

**Reply by metalsadman** [reply]
make a codepen

**Comment by jaymc-arg** [comment, marked as answer]
Solved!
When creting the codepen it indeed worked, so I checked my code and I realized I was using a v-model to bind the selected object that was `ref([])` and it had to be `ref(null)`, so the `can not read '0' of undefined` was from that object.
Thank you all!
